src/test/java/primerTest/LoginTests.java

```diff
@@ -0,0 +1,106 @@
+
+        Assert.assertTrue(usernameInput.isDisplayed());
+        usernameInput.sendKeys("tomsmith");
+        Assert.assertTrue(passwordInput.isDisplayed());
+        passwordInput.sendKeys("SuperSecretPassword!");
+        Assert.assertTrue(buttonInput.isDisplayed());
+        buttonInput.click();
+        utilities.waitSeconds(2);
+
+        var greenframeVisible = driver.findElement(By.id("flash"));
+        var logoutInput = driver.findElement(By.cssSelector("i[class='icon-2x icon-signout']"));
+
+        logs.debug("verificando el cuadro verde");
+        Assert.assertTrue(greenframeVisible.isDisplayed());
+        logs.debug("verificando el botón este visible");
+        Assert.assertTrue(logoutInput.isDisplayed());
+        logs.debug("Haciendo logout click");
+        logoutInput.click();
+
+        usernameInput = driver.findElement(By.id("username"));
+        Assert.assertTrue(usernameInput.isDisplayed());
+
+        logs.info("Realizando el test Regression -> test 2");
+
+        usernameInput = driver.findElement(By.id("username"));
+        Assert.assertTrue(usernameInput.isDisplayed());
+        passwordInput = driver.findElement(By.id("password"));
+        Assert.assertTrue(passwordInput.isDisplayed());
+        buttonInput = driver.findElement(By.className("radius"));
+        usernameInput.sendKeys("hola123");
+        passwordInput.sendKeys("heheheeh");
+        buttonInput.click();
+        utilities.waitSeconds(2);
+        var redframeInput = driver.findElement(By.id("flash"));
+        Assert.assertTrue(redframeInput.isDisplayed());
+    }
+
+    @Test(groups = {regression})
+    public void exercise3() {
+        logs.info("Realizando ejercicio 3");
+        utilities.waitSeconds(2);
+        var checkBox1Input = driver.findElements(By.cssSelector("input[type='checkbox']"));
+        List<WebElement> checkboxList = checkBox1Input;
+
+        var firstCheckBox = checkboxList.get(0);
+        var secondCheckBox = checkboxList.get(1);
+
+        Assert.assertFalse(firstCheckBox.isSelected());
+        Assert.assertTrue(secondCheckBox.isSelected());
+        firstCheckBox.click();
+        secondCheckBox.click();
+
+        utilities.waitSeconds(3);
+        Assert.assertTrue(firstCheckBox.isSelected());
+        Assert.assertFalse(secondCheckBox.isSelected());
+    }
+
+    @Test(groups = {regression})
+    public void exercise4() {
+        logs.info("Realizando ejercicio 4");
+        utilities.waitSeconds(2);
+        var buttonAddInput = driver.findElement(By.xpath("//button[text()='Add Element']"));
+        for (var i = 0; i <= 9; i++) {
+            buttonAddInput.click();
+        }
+        utilities.waitSeconds(5);
+        var buttonDeletes = driver.findElements(By.className("added-manually"));
+        List<WebElement> buttonDeleteList = buttonDeletes;
+        Assert.assertEquals(buttonDeleteList.size(), 10);
+        for (var i = 0; i <= 9; i++) {
```

Comment:
innecesario el for convenciona, usa foreach

Reply:
resuelto